index.js

```diff
@@ -398,13 +400,18 @@ class Batch {
     this.root = null
     this.length = 0
     this.options = options
+    this.locked = null
     this.onseq = this.options.onseq || noop
   }
 
   ready () {
     return this.tree.ready()
   }
 
+  async lock () {
```

Comment:
Should this be public on the batch?

Reply:
ya, needs docs in the future (for eagerly aquiring the lock)